storage/src/mmr/mem.rs

```diff
@@ -359,220 +359,225 @@ mod tests {
+        "e00236775485cb37e90d95b78c41773c4a1f0ea144ea79f48fcc035371002a43",
+        "68bcd7fb41112e1af3c83017c7365c512f7fbf0d5021a025f6f1b7017a0c21e7",
+        "868fdb9fa253928ae6edea1ab37d5dc7543ac68fe063f2aa54ade5f27374018f",
+        "d2603d86737d42200e8cc26911e88ff5dd7ae0b26d4f3db005e64a20c1c145ba",
+        "284cd27321a9ea5ff9b0121254a2b5a97a94d0285b3f4feca0ff0daae2e608c3",
+        "a0ef34637f25035c8d9b751df77afeacd4e96fc361cd022dfd430ba60541a830",
+        "789364e8064c04588c308bc0b7f8245c052daaaa3c2465484ce2c861e83b323c",
+        "dfd43f820de855c96b68275b17707f96132bbd0142e0752c947d134a96ca9e02",
+        "599f47ed2b528f8e69a869a69321c26ff5d381a07450d1e52ea541b3420a3700",
+        "6182cdd179e500fabc42b628e00a3f10e92f886a3f257f2851cd6ef88be5983c",
+        "0c935ea6d867d6ad94afad712fe8ab774b1d0a237a0ee6761c9dbe9d064540ad",
+        "55a59168bd4fc73d33f07aa753f96498d0b5e2d44fcd9ee9842618d34ebd313f",
+        "e0d63511aa1d70455e3222aca71b071ba9e8a841528bdb1494724a0b660a6d21",
+        "5bc1f5d00d9ea14e60b1e30ba7dc4a4df63d0b4161ab76b9245faf865069b601",
+        "9e5effda000fc36b9f9d7ccfcb1c0fa80633ba0c4e1860921615a3373a233925",
+        "d9843b92564ed5732eada545b1b6cee8ad33e0d5b8c076e84d3719d82d6b3631",
+        "d963b95dd975b603e0795465e18d215718f93d9452996115f44d6e98bf637c17",
+        "0168affb6e415c4e32a43a006c56b0f04178b9dc4c0776f01254dbd390cee20a",
+        "867b018e2ea8896500c26c098e1264454d5ba7f3cf540e035efef7058cf22925",
+        "197578c75dedcf8e2ad69cb2b54cc3199d624ff5879640be720fee6fe25127d1",
+        "233a3563f0e4ad1f9167794d8432440343a4f7c0ee884e637786a9dbfadc2e28",
+        "d43d51330ef1802d96f45eae4774b230609f0dd8c36766b525f4c63e3d240ec6",
+        "142a320ae73fee77266d8102e2de8d8aadedeb77dd88452b152e82ec7d19b7b6",
+        "0bc659753f0a0c45b8d235ee2b5684067537951df018fe66d0b0c7e2d725fa2f",
+        "40caf9a8301eb5a39ae293c04130073bd9056c025678032055b4a1c4f71390ed",
+        "8ed975b21802a84795917b54275aa2f66c58817151a210de441252acb7911dee",
+        "d6583949a6187fe87caaf426c04bcd4eae291d9e0fb2e1d4873cb3b2ed7e3cb9",
+        "2bcd0829fd8b0e89841a2110baf09008cafde1fede304100c57a897753724880",
+        "71e9170d6c7d8d85b7d9331305ff27b6a6f8a10fa7d145f9f42adae04b776319",
+        "b0aa21e3c0e24118316eebf64d318c448adb9e7772aa5bf92c57a261aaed15f3",
+        "8e7ef9a87c2dbc1f5eaa6d0c37ff0d4334c94b8d35e055a643e978336cee2207",
+        "9da033b7451c9ed5a24dae924170d752fad0920681646745c801a119269da5a2",
     ];
 
     /// Test that the MMR root computation remains stable by comparing against previously computed
     /// roots.
     #[test]
     fn test_root_stability() {
-        let mut hasher = Sha256::new();
-        let mut mmr = Mmr::<Sha256>::new();
         for i in 0..200 {
+            let mut hasher = Sha256::new();
+            let mut mmr = Mmr::<Sha256>::new();
             for _ in 0u64..i {
                 let element = hash(&i.to_be_bytes());
```

Comment:
If you already have a hasher, you might as well do `update()` and `finalize()` (each call to `hash` spawns a new hasher)

Reply:
thx, done.